lib/src/rules/prefer_asserts_in_initializer_lists.dart

```diff
@@ -101,8 +101,8 @@ class _AssertVisitor extends RecursiveAstVisitor {
   bool _hasMethod(MethodElement element) {
     final type = classElement.type;
     final name = element.name;
-    return type.lookUpMethod(name, element.library) == element ||
-        type.lookUpInheritedMethod(name) == element;
+    return element == type.lookUpMethod(name, element.library) ||
```

Comment:
The `==` impl here is not symmetric, which is pretty annoying.

CC @scheglov

Reply:
CC @bwilkerson

Reply:
Yeah, this stinks.

The reason why it is not symmetrical is that `type.lookUpMethod` might return `MethodMember`, which is `MethodElement` with type information (i.e. type parameters replaced with specific type arguments from the `type`). And `MethodMember` or `Member` at all does not implement `operator==`. It is not obvious what `==` means for members. Is it just comparison of base elements? Or also types of the member and the given element?